### DISCUSSION TITLE
RetroArch limited support

### DISCUSSION THREAD

**Discussion by FlotterCodername** [retronas/discussions#101] [Ideas]
https://github.com/danmons/retronas/issues/1#issuecomment-1031862489 from @gingerbeardman gave me an idea on how some limited RetroArch support can be achieved. That is vanilla RetroArch. Frontends over the top of RA may run completely different CFGs which they specify on a case-by-case basis via the CLI.

User uploads their own `retroarch.cfg` for modification, runs a task/service and gets a modified version back that lightly integrates RetroNAS.
### `retroarch.cfg` basic format:
- No section headers
- key_name = "value"
- Always dquotes regardless of data type.
- `#` starts line comment

To make this cleaner, the `#include cfg_name.cfg` statement could be used. So in the original retroarch.cfg, you'd only add at the bottom:
```
#include "retroarch-retronas.cfg"
```
Problem with that: this include statement will be flushed (along with any comments) if the user has `config_save_on_exit = "true" ` 🥴.
So I don't know if that's what you'd want there.

On Windows, the file will be expected next to the original  `retroarch.cfg`. I'm not familiar with macOS/Linux RA, sorry.

The relevant config keys:
```
# See above
config_save_on_exit = "true", "false"

# Should probably make sure the overrides are exposed to user
quick_menu_show_save_content_dir_overrides = "true"

# Cannot browse/set SMB shares from Windows GUI, only set them in config. So make sure the setting sticks:
use_last_start_directory = "false"

# This notation works on Windows. Backslash needs no escaping.
rgui_browser_directory = "\\retronas\share_name"                # It's not just RGUI

# System = BIOS and similar
system_directory = <path>                                       # default value :\system
# All "in_content_dir" settings override the related explicit setting
systemfiles_in_content_dir = "true", "false"
# More stuff
savefile_directory = <path>                                     # default value :\saves
savefiles_in_content_dir = "true", "false"
savestate_directory = <path>                                    # default value :\states
savestates_in_content_dir = "true", "false"
screenshot_directory = <path>                                   # default value :\screenshots
screenshots_in_content_dir = "true", "false"

# The following have no content dir override afaik
# Core options (maybe useful for some users?)
rgui_config_directory = <path>                                  # default value :\config; It's not just RGUI
# Boxarts etc.
thumbnails_directory = <path>                                   # default value :\thumbnails
```
There are more directories that can be configured, but I limited it to the ones that make the most sense.

**Comment by sairuk** [comment]
I'd like to see where this goes, but I won't particularly have any valuable input on RA since I haven't been a user of it for many, many years. 

For reference, unless this has changed in the last 3yrs since I last looked at RA, RA on Linux puts the` retroarch.cfg` file under `$HOME/.config/retroarch/`

**Comment by gingerbeardman** [comment]
I'm glad my comment gave you an idea, but I have to say it's not clear what you are proposing? Some kind of pre-processor?
